### PR TITLE
[Chat][RTE] Error bar update for Rich Text Editor

### PR DESCRIPTION
**Pull request by vhuseinova-msft**
# What

- Updated functionality for error bar in RTE 
- Added tests
- Added props for Storybook RTE page

# Why

# How Tested

Tests, storybook

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/RTE/RTESendBox.tsx [hunk 1]
```diff
@@ -61,24 +68,76 @@ export interface RTESendBoxProps {
...
 
   const sendMessageOnClick = (): void => {
     if (disabled || contentValueOverflow) {
       return;
     }
+    // Don't send message until all files have been uploaded successfully
+    /* @conditional-compile-remove(file-sharing) */
+    setFileUploadsPendingError(undefined);
+
+    // /* @conditional-compile-remove(file-sharing) */
```
Comment (vhuseinova-msft) [inline packages/react-components/src/components/RTE/RTESendBox.tsx:111]: Thie will be added later. Added and comment here, so we won't forget about it

File: packages/react-components/src/components/utils/SendBoxUtils.ts [hunk 2]
```diff
@@ -22,3 +28,21 @@ export const hasIncompleteFileUploads = (activeFileUploads: ActiveFileUpload[] |
 export const hasCompletedFileUploads = (activeFileUploads: ActiveFileUpload[] | undefined): boolean => {
   return !!activeFileUploads?.find((file) => !file.error);
 };
+
```
Comment (palatter) [inline packages/react-components/src/components/utils/SendBoxUtils.ts:31]: Thanks for making these common!

**Review by palatter — APPROVED** [review]

File: packages/react-components/src/components/RTE/RTESendBox.tsx [hunk 3]
```diff
@@ -53,6 +59,10 @@ export interface RTESendBoxProps {
    * @beta
    */
   onCancelFileUpload?: (fileId: string) => void;
+  /**
+   * Optional override behavior on send button click
```
Comment (emlynmac) [inline packages/react-components/src/components/RTE/RTESendBox.tsx:63]: The comment here implies that the message gets sent regardless of whether this is set(implemented).
Is this callback really used to implement the send button functionality?

Comment (vhuseinova-msft) [reply packages/react-components/src/components/RTE/RTESendBox.tsx:63]: I'm not sure that I agree.
The comment specify that this callback will change the behaviour for a send button click event, the default behavious is that clicks don't cause any action for sending. 

Just wanted to add that the same pattern is reused from SendBox and I see that we've also used the same pattern for other callbacks, like cancelling file uploads. And some React/HTML events that don't have any default behavior.  

But I'm open for any suggestions for the name of this callback

Comment (emlynmac) [reply packages/react-components/src/components/RTE/RTESendBox.tsx:63]: My understanding is that the callback is called when we want to send the message contents.
If this is the case, I'd like to see the description be more like:

`Function called when the send message button is pressed`

If the original SendBox also has this description, maybe that too should be updated?

**Review by emlynmac — APPROVED** [review]

File: packages/react-components/src/components/RTE/RTESendBox.tsx (hunk 3 above)
Comment (vhuseinova-msft) [reply packages/react-components/src/components/RTE/RTESendBox.tsx]: Made the callback required and updated the comment as agreed

**Pull request merged**